### PR TITLE
Use doctest to update pydotprint plots and debugprint output

### PR DESCRIPTION
**Pull request by cangermueller**
Updates documentation with doctest test cases to automatically a) generate and visualize pydotprint graphs and b) generate `pprint` and `debugprint` output. Executing `doc/scripts/docgen.py` will generate plots automatically. See #2711
* Updates `doc/tutorial/symbolic_graphs.txt`
* Update `cifarsc2011/advanced_theano.txt`
* Substitutes `pydotprint_variables` by `pydotprint` in `hpcs2011_tutorial/logreg_example.py`

### CODE REVIEW

**Comment by nouiz** [comment]
doc/cifarSC2011/pics/logreg_pydotprint_predic.png

this file seem to be a duplicate of doc/cifarSC2011/pics/logreg_pydotprint_predict.png

that is a left over from your development.

File: doc/cifarSC2011/advanced_theano.txt [hunk 1]
```diff
@@ -275,62 +275,99 @@ Exercise 5
...
- | |TensorConstant{0.5} [@181771148]
->>> theano.printing.debugprint(predict) # doctest: +SKIP
-Elemwise{Composite{neg,{sub,{{scalar_sigmoid,GT},neg}}}} [@183160204] ''   2
- |dot [@183018796] ''   1
- | |x [@183000780]
- | |w [@183000812]
- |InplaceDimShuffle{x} [@183133580] ''   0
- | |b [@183000876]
- |TensorConstant{[ 0.5]} [@183084108]
+>>> theano.printing.debugprint(prediction) # doctest: +NORMALIZE_WHITESPACE
```
Comment (nouiz) [inline doc/cifarSC2011/advanced_theano.txt]: I would add a comment to the user to tell that this is the graph build by the user, it isn't optimized.

File: doc/cifarSC2011/advanced_theano.txt [hunk 2]
```diff
@@ -275,62 +275,99 @@ Exercise 5
...
+    |       |Elemwise{neg,no_inplace} [@I] ''
+    |       | |dot [@J] ''
+    |       |   |x [@K]
+    |       |   |w [@L]
+    |       |DimShuffle{x} [@M] ''
+    |         |b [@N]
+    |DimShuffle{x} [@O] ''
+      |TensorConstant{0.5} [@P]
+
+>>> theano.printing.debugprint(predict) # doctest: +NORMALIZE_WHITESPACE
```
Comment (nouiz) [inline doc/cifarSC2011/advanced_theano.txt]: I would add a comment to tell that this is the same as the graph above but after optimization.

File: doc/cifarSC2011/advanced_theano.txt [hunk 3]
```diff
@@ -275,62 +275,99 @@ Exercise 5
...
+     |InplaceDimShuffle{x} [@I] ''   0
+     | |b [@J]
+     |TensorConstant{(1,) of 0.5} [@K]
+
 
 - Picture Printing of Graphs
+``pydotprint`` requires graphviz and pydot.
 
->>> theano.printing.pydotprint_variables(prediction) # doctest: +SKIP
+>>> theano.printing.pydotprint(prediction, outfile="pics/logreg_pydotprint_prediction.png", var_with_name_simple=True)
```
Comment (nouiz) [inline doc/cifarSC2011/advanced_theano.txt]: idem

File: doc/cifarSC2011/advanced_theano.txt [hunk 4]
```diff
@@ -275,62 +275,99 @@ Exercise 5
...
+The output file is available at pics/logreg_pydotprint_prediction.png
 
-.. image:: ../hpcs2011_tutorial/pics/logreg_pydotprint_prediction.png
+.. image:: ./pics/logreg_pydotprint_prediction.png
    :width: 800 px
 
-All pydotprint* requires graphviz and pydot
+>>> theano.printing.pydotprint(predict, outfile="pics/logreg_pydotprint_predict.png", var_with_name_simple=True)
```
Comment (nouiz) [inline doc/cifarSC2011/advanced_theano.txt]: idem

File: doc/cifarSC2011/advanced_theano.txt [hunk 5]
```diff
@@ -275,62 +275,99 @@ Exercise 5
...
+The output file is available at pics/logreg_pydotprint_predict.png
 
->>> theano.printing.pydotprint(predict) # doctest: +SKIP
-
-.. image:: ../hpcs2011_tutorial/pics/logreg_pydotprint_predic.png
+.. image:: ./pics/logreg_pydotprint_predict.png
    :width: 800 px
 
->>> theano.printing.pydotprint(train) # This is a small train example! # doctest: +SKIP
+>>> theano.printing.pydotprint(train, outfile="pics/logreg_pydotprint_train.png", var_with_name_simple=True)
```
Comment (nouiz) [inline doc/cifarSC2011/advanced_theano.txt]: I would tell that this is the optimized graph.

**Comment by nouiz** [comment]
I have a few small comment to make it even better. Thanks for the update.

Just to be sure, if I run python doc/scripts/docgen.pdf, do this execute the doctest?

**Comment by cangermueller** [comment]
@nouiz, I incorporated your comments and checked the html files generated by `scripts/docgen.py`.

**Comment by nouiz** [comment]
thanks. I'll merge this PR now. Is the ticket related to this PR completed or there is other stuff left?

**Comment by cangermueller** [comment]
#2846 updates the remaining tutorials mentioned in #2711, which can closed now.

**Pull request merged**